sycl/include/CL/sycl/accessor.hpp

```diff
@@ -834,6 +834,9 @@ class accessor :
 #endif // __SYCL_DEVICE_ONLY__
 
     size_t Result = 0;
+#ifdef __SYCL_DEVICE_ONLY__
+#pragma unroll
```

Comment:
@erichkeane , could you, please, clarify if it will work out in device compilation mode?

Reply:
Does this diagnose the unknown pragma (which may break -Werror builds) when SYCL headers are consumed by a compiler other than Clang, like MSVC?

Reply:
Yes, it breaks werror build. That's why I've added device check.